util/flightcontrol/flightcontrol.go

```diff
@@ -27,12 +27,29 @@ type Group struct {
 	m  map[string]*call // lazily initialized
 }
 
+type retries struct {
+	num     int
+	backoff time.Duration
+}
+
 // Do executes a context function syncronized by the key
 func (g *Group) Do(ctx context.Context, key string, fn func(ctx context.Context) (interface{}, error)) (v interface{}, err error) {
+	retries := retries{backoff: time.Second}
```

Comment:
This is way too big. Should start from `Gosched`, then backoff starting from 1 millisecond.